### PR TITLE
Store properties injected by the native app

### LINKED ISSUE #28: Store Tokens and Keys
# Why

As we are passing tokens and keys from the main app. We need to store them somehow in the react environment so they can be accessed where needed.

### PR DESCRIPTION
**Pull request by Ecarrion**
closes #28 

# Why

This PR locally stores the `BlogIf` and `token` properties that are injected by the native app. This in order to avoid having to pass these properties as parameters through all the react app functions.

# How

- Adds `AsyncStorage` react dependency.

- Adds `storeDependency()` `readDependency()` functions to store the properties injected from the native app.

- Update `index.tsx` to store the injected properties in the local storage.

- Update the `jetpackFetch()` function to read the `blogId` and `token` properties from the local storage.

# Testing steps

- Run the app and make sure the shipping list loads correctly (like before).

### CODE REVIEW

File: yarn.lock
Comment (wzieba) [inline yarn.lock]: I'm uncertain if it was an intended change - I think we should decide on using `npm` or `yarn`. I think we should keep with `yarn` to keep the current setup and consistency.

So, what I've noticed, is that `npm` would change `yarn.lock` to its coordinates if used. I'd suggest reverting changes from `yarn.lock` and use `yarn add @react-native-async-storage/async-storage --save`

**Comment by wzieba** [comment]
@Ecarrion which app have you been using to test this change? iOS or Android? I have some difficulties with the Android build and I don't know how to link React Native to the WooCommerce-iOS app.

**Comment by wzieba** [comment]
I think that `@react-native-async-storage/async-storage` dependency suffers from the same issues as `@react-navigation/native` does in Android.

#27 tries to address it, but for now, the Android app won't build.

**Review by wzieba — APPROVED** [review]
The code looks good! 👍 

After merging #27 and adding changes in https://github.com/woocommerce/woocommerce-android/pull/9225 , I was able to successfully run the app and storing credentials works well.

Under the hood, when it comes to Android, values are stored in SQLite database: 

image

I wonder why the authors of `async-storage` decide to do this, instead of `SharedPreferences`.

File: yarn.lock
Comment (wzieba) [reply yarn.lock]: I allowed myself to address this comment here: [99e0359](https://github.com/woocommerce/WooCommerce-Shared/pull/31/commits/99e0359de3fdc6e5b53ea53c699f2ac8de17f0e2)

**Comment by wzieba** [comment]
I've just notice that when the screen is opened, JS loggs:

```
 LOG  Error storing api-token [Error: [AsyncStorage] Passing null/undefined as value is not supported. If you want to remove value, Use .removeItem method instead.
Passed value: undefined
Passed key: api-token
]
```

Does it happen for you too @Ecarrion ?

File: yarn.lock
Comment (Ecarrion) [reply yarn.lock]: Thanks for this @wzieba. My bad, I did `npm install ...` and didn't think much about it. I'll use `yarn` from now on.

**Review by wzieba — APPROVED** [review]

**Pull request merged**
